src/main/org/nlogo/window/ButtonWidget.scala

```diff
@@ -374,15 +373,22 @@ class ButtonWidget(random:MersenneTwisterFast) extends JobWidget(random)
     def paintButtonRectangle(g: Graphics) {
       g.setColor(getPaintColor)
       g.fillRect(0, 0, getWidth(), getHeight())
-      def renderImages(g: Graphics, dark: Boolean) {
+      def renderImages(g: Graphics, inverse: Boolean) {
         def maybePaintForeverImage() {
           if (forever) {
-            val image = if (dark) FOREVER_GRAPHIC_DARK else FOREVER_GRAPHIC
+            val image = new ImageIcon(java.awt.Toolkit.getDefaultToolkit.createImage(
```

Comment:
I'm not necessarily saying that all images need to be doubled. I imagine that links and patches probably won't be. However, this code only affects the forever image on a forever button. So the choice is one extra image vs an anonymous RGBImageFilter. The image is both simpler and more flexible.